### PR TITLE
Dependency graph viewer README

### PR DESCRIPTION
**Pull request by iserrato**
README to help devs use the DependencyGraphViewer tool

### CODE REVIEW

**Review by sbomer — COMMENTED** [review]
I think we should combine this doc with https://github.com/dotnet/runtime/blob/9d6396deb02161f5ee47af72ccac52c2e1bae458/docs/workflow/debugging/coreclr/debugging-compiler-dependency-analysis.md - either move the new contents there, or extract out the shared bits into a new file.

File: src/coreclr/tools/aot/DependencyGraphViewer/DependencyGraphsForm.cs [hunk 1]
```diff
@@ -69,8 +69,8 @@ private void help_Click(object sender, EventArgs e)
                                     Dependency Graph Viewer
 
                                     This application allows viewing the dependency graph produced by
-                                    the AOT compilation. Graphs can be generated by running the compiler
-                                    or loading a .dgml file.
+                                    the AOT compilation or the Linker. Graphs can be generated by running
+                                    the AOT compiler or loading a .dgml file.
```
[the remark this answers is not in the document]
Comment (iserrato) [reply src/coreclr/tools/aot/DependencyGraphViewer/DependencyGraphsForm.cs:73]: What I meant to say is that opening a dgml creates a graph in the program that can then be explored, but I can cut it

File: src/coreclr/tools/aot/DependencyGraphViewer/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Why use this tool?
+If you are developing with NativeAOT or using the Linker, you might ask yourself: why are types or methods being kept after trimming? Sometimes, it can be unclear why certain types or methods are retained after a self-contained application is trimmed.
+
+## How to build and run
+1. Launch Visual Studio
+2. Load `DependencyGraphViewer.sln`
+3. Build and Run as normal
+## Using ETW logs
+Visual Studio must be run as an administrator if collecting ETW events when compiling with NativeAOT.
```
[the remark this answers is not in the document]
Comment (iserrato) [reply src/coreclr/tools/aot/DependencyGraphViewer/README.md:11]: gotcha, I'll clarify that

**Review by sbomer — APPROVED** [review]

**Comment by iserrato** [comment]
No functional changes

**Pull request merged**